lib/logstash/codecs/compress_spooler.rb

```diff
@@ -21,13 +21,13 @@ def decode(data)
   end # def decode
 
   def encode(event)
+    # use normalize to make sure returned Hash is pure Ruby for
+    # MessagePack#pack which relies on pure Ruby object recognition
+    @buffer << LogStash::Util.normalize(event.to_hash).merge(LogStash::Event::TIMESTAMP => event.timestamp.to_iso8601)
+    # If necessary, we flush the buffer and get the data compressed
```

Comment:
I did not thought about it, so let me understand you are saying we should be sure not to output internal variables like ```@data``` to the buffer? isn't?